flax/linen/module.py

```diff
@@ -96,7 +96,7 @@ def _module_repr(module: 'Module', num_spaces: int = 4):
     rep += '# attributes\n'
     for attr in attributes.keys():
       # TODO(jheek): can we get a nice string representation of attribute types?
-      value = getattr(module, attr, None)
+      value = module.__dict__.get(attr, None)
```

Comment:
Will probably depend on the context, in this case it's probably the right thing to do as we are getting the local attributes, and more than that, it is necessary to avoid certain issues when getting the representation for bounded modules, see https://github.com/google/flax/commit/a648605bd708a19c9ce8bc332fa359cb98585837 which @chiamp recently merged.